sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java

```
@@ -0,0 +1,134 @@
+    JdkHttpResponseSync(final HttpRequest request, int statusCode, HttpHeaders headers, byte[] bytes) {
+        super(request, statusCode, headers);
+        this.bodyStream = null;
+        this.bodyBytes = bytes;
+    }
+
+    JdkHttpResponseSync(final HttpRequest request, java.net.http.HttpResponse<InputStream> streamResponse) {
+        super(request, streamResponse.statusCode(), fromJdkHttpHeaders(streamResponse.headers()));
+        this.bodyStream = streamResponse.body();
+        this.bodyBytes = null;
+    }
+
+    @Override
+    public Flux<ByteBuffer> getBody() {
+        if (bodyBytes != null) {
+            return Mono.fromSupplier(() -> ByteBuffer.wrap(bodyBytes)).flux();
+        } else {
+            return FluxUtil.toFluxByteBuffer(bodyStream).doFinally(ignored -> close());
+        }
+    }
+
+    @Override
+    public Mono<byte[]> getBodyAsByteArray() {
+        if (bodyBytes != null) {
+            return Mono.just(bodyBytes);
+        } else {
+            return super.getBodyAsByteArray();
+        }
+    }
+
+    @Override
+    public BinaryData getBodyAsBinaryData() {
+        if (bodyBytes != null) {
+            return BinaryData.fromBytes(bodyBytes);
+        } else {
+            // we shouldn't create multiple binary data instances for a single stream
+            return getBinaryData();
+        }
+    }
+
+    @Override
+    public void writeBodyTo(WritableByteChannel channel) throws IOException {
+        if (bodyBytes != null) {
+            channel.write(ByteBuffer.wrap(bodyBytes));
+        } else {
+            try {
+                int nRead;
+                byte[] data = new byte[STREAM_READ_SIZE];
+                while ((nRead = bodyStream.read(data, 0, data.length)) != -1) {
+                    channel.write(ByteBuffer.wrap(data, 0, nRead));
+                }
+            } catch (IOException ex) {
+                throw LOGGER.logExceptionAsError(new UncheckedIOException(ex));
+            }
```

nit: method is checked with IOException, just let it be thrown

---

it's not checked in `HttpResponse` and I have to wrap it here

---

It looks checked to me? https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/HttpResponse.java#L163